deepchem/feat/molecule_featurizers/mat_featurizer.py

```diff
@@ -0,0 +1,67 @@
+from deepchem.feat.base_classes import MolecularFeaturizer
+from deepchem.utils.molecule_feature_utils import one_hot_encode
+try:
+  import logging
+  import numpy as np
+  import pandas as pd
+  import torch
+  from rdkit import Chem
+  from rdkit.Chem import AllChem
+  from rdkit.Chem import MolFromSmiles
+  from sklearn.metrics import pairwise_distances
+  from torch.utils import datasets
+except:
+  raise ImportError("Required Modules not found.")
+
+class MATFeaturizer(MolecularFeaturizer):
+
+  def __init__(
+      self,
+      add_dummy_node=True,
+      one_hot_formal_charge=True,
+  ):
+    self.mol = mol
+    self.add_dummy_node = add_dummy_node
+    self.one_hot_formal_charge = one_hot_formal_charge
+
+  def atom_features(self, atom, one_hot_formal_charge=True):
+    attrib = []
+    attrib += one_hot_encode(atom.getAtomicNum(),
+                       [5, 6, 7, 8, 9, 15, 16, 17, 35, 53, 999])
+    attrib += one_hot_encode(len(atom.GetNeighbors()), [0, 1, 2, 3, 4, 5])
+    attrib += one_hot_encode(atom.GetTotalNumHs(), [0, 1, 2, 3, 4])
+
+    if one_hot_formal_charge:
+      attrib += one_hot_encode(atom.GetFormalCharge(), [-1, 0, 1])
+    else:
+      attrib.append(atom.GetFormalCharge())
+
+    attrib.append(atom.IsInRing())
+    attrib.append(atom.GetIsAromatic())
+
+    return np.array(attrib, dtype=np.float32)
+
+  def _featurize(self, mol, add_dummy_node, one_hot_formal_charge):
+    node_features = np.array(
```

Comment:
`add_dummy_node` and `one_hot_formal_charge` are already class attributes. You can just use `self.add_dummy_node` instead?

Reply:
Done!